### PR TITLE
Adding selected measure and dimension to the state

### LINKED ISSUE #1905: refine semantics & behavior of protobuf-based URL
A few pieces of state that need to make their way into the URL:
- the "focus metric" – this is the metric that is on the leaderboards & is the focus column in the dimension table.
- dimension table view – either we handle this in _routing_, or we handle it in _state_. Handling it in routing is a bit more complex but probably doable. To do this in the route, we would have to ensure that the screen doesn't redraw when the user navigates to the dimension table. Doing this in state is easier in the short-run so I would opt to do that, & we can deprecate that later.
- include/exclude state should update based on the selected filter (in the future the actual state regardless of selected filter should be stored)
- state – the back button should trigger the right navigation change.

There may be other parts of state that we need to add but these are the obvious ones.

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #1905

### CODE REVIEW

**Review by hamilton — CHANGES_REQUESTED** [review]
Thanks for the PR!

### bits of state
All the additional state pieces appear to work as needed. I made a suggestion to rename `time_granularity` to `time_grain` in the dash since that's what we actually are calling it. Probably the right time to align on the terminology.


### back button dynamics
Unfortunately, the back button only sort of works with this PR. The forward button is not preserved and I have to click twice to make the back button take effect. We have two options to fix this:

*use the history API to make sure that the actual querystring state is navigable with a back + forward button.* This will require some manual work to make sure it works as expected. It also means we need to de-duplicate history here, since I think that's what's happening w/ the current behavior where I need to click twice to have it take effect.

*Move to a pure route-based version of the protobuf message*, and version it, using sveltekit's routing to take care of the capture. After chatting with @medriscoll, it seems like it might make more sense for us to _embed the protobuf into the route itself_ rather than rely on querystrings. 

The URL would then look like `<base>/dashboards/<dashboardName>/v1--<protobuf>`. Taking this approach means we could probably version in a specific route. What we get out of this is that in theory the back / forward button works by default. It would mean we have to create some routes & get the logic right, but it will surely be more robust than trying to get the history semantics right. See [advanced routing docs](https://kit.svelte.dev/docs/advanced-routing) for more information. I think this means the route itself would look like `/dashboards/[name]/v1--[state]`. We would have to figure out:
- what happens when no filters are selected. I think we still keep `/dashboards/[name]` since it's identical to `/dashboards/[name]/v1--[state]` with the exception of `[state]`.
- when I go from a no-filter selected -> a filter selected (or some state changed), make sure it doesn't completely refresh the page.

---

Happy to talk about how we handle this tomorrow if that'll help.

File: web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte [hunk 1]
```diff
@@ -264,7 +264,9 @@
         formatPreset: m.format as NicelyFormattedTypes,
       };
     });
-    values = humanizeGroupByColumns(values, measureFormatSpec);
+    if (measureFormatSpec) {
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte:267]: UI broke without this check

File: proto/rill/ui/v1/dashboard.proto [hunk 2]
```diff
@@ -5,14 +5,22 @@ import "google/protobuf/timestamp.proto";
...
 import "rill/runtime/v1/queries.proto";
 
+// DashboardState represents the dashboard as seen by the user
 message DashboardState {
+  // Selected time rage
   DashboardTimeRange time_range = 1;
+  // Dimension filters applied
   rill.runtime.v1.MetricsViewFilter filters = 2;
+  // Selected time granularity
+  rill.runtime.v1.TimeGrain time_granularity = 3;
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply proto/rill/ui/v1/dashboard.proto]: Merged main and fixed

File: web-common/src/proto/gen/rill/ui/v1/dashboard_pb.ts [hunk 3]
```diff
@@ -9,19 +9,46 @@ import { MetricsViewFilter } from "../../runtime/v1/queries_pb.js";
 import { TimeGrain } from "../../runtime/v1/catalog_pb.js";
 
 /**
+ * DashboardState represents the dashboard as seen by the user
+ *
  * @generated from message rill.ui.v1.DashboardState
  */
 export class DashboardState extends Message<DashboardState> {
   /**
+   * Selected time rage
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/proto/gen/rill/ui/v1/dashboard_pb.ts]: Merged main and regenerated

**Comment by hamilton** [comment]
@AdityaHegde fixed a few things in this PR:
- the charts now tween on back button. There was an intermediate step where you were setting the `startValue` and `endValue` to `undefined` without checking first to see if there were valid dates to set.
- I fixed a weird render bug where if I select items in the leaderboard, for some reason they were not propagating the active state on back / forward button.

File: proto/rill/ui/v1/dashboard.proto [hunk 4]
```diff
@@ -5,14 +5,22 @@ import "google/protobuf/timestamp.proto";
 import "rill/runtime/v1/catalog.proto";
 import "rill/runtime/v1/queries.proto";
 
+// DashboardState represents the dashboard as seen by the user
 message DashboardState {
+  // Selected time range
   DashboardTimeRange time_range = 1;
+  // Dimension filters applied
   rill.runtime.v1.MetricsViewFilter filters = 2;
+  // Selected time granularity
```
Comment (hamilton) [inline proto/rill/ui/v1/dashboard.proto:14]:
```suggestion
  // Selected time grain
```

**Review by hamilton — APPROVED** [review]
lgtm. I added a few changes to the styling to make it work correctly.

We'll need to add tests for `fromProto` / `toProto`, but given that we need this PR to land before we can finish #1942, let's get this in.

**Pull request merged**